File: flutter/lib/domain/island/owned_island.dart
import '../space/coord_2d.dart';
import 'island.dart';

class OwnedIsland implements Island {
  @override
  final int id;
  @override
  final Coord2D coordinate;
  @override
  final int radius;
  final int incomePerHour;
  final String uid;
  // constructor
  OwnedIsland({required this.id, required this.coordinate, required this.radius, required this.incomePerHour, required this.uid});
}

File: jvm/ship-conquest/src/main/kotlin/pt/isel/shipconquest/domain/world/islands/OwnedIsland.kt
package pt.isel.shipconquest.domain.world.islands

import pt.isel.shipconquest.domain.space.Vector2
import java.time.Instant

data class OwnedIsland(
    override val islandId: Int,
    override val coordinate: Vector2,
    override val radius: Int,
    val uid: String,
    val incomePerHour: Int,
    val conquestDate: Instant,
    val ownershipDetails: OwnershipDetails
): Island